plugin/mock_proto/generate.go

```diff
@@ -0,0 +1,3 @@
+//go:generate mockgen -destination mock.go github.com/hashicorp/terraform/plugin/proto ProviderClient,ProvisionerClient,Provisioner_ProvisionResourceClient,Provisioner_ProvisionResourceServer
```

Comment:
Is this `mockgen` something that our `make tools` target should install?

Reply:
oh yes, I forgot there's a `generate` make target. I also haven't vendored `gomock` yet.